entities/library.rb

```diff
@@ -0,0 +1,62 @@
+require_relative '../dependency'
+require_relative '../data/data_generator'
+
+class Library
+  include DataGenerator
+  include DataUtilities
+  include Statistics
+
+  attr_reader :authors, :books, :readers, :orders
+
+  def add_objects(*objects)
+    objects.each do |object|
+      case object.class
+      when Author
+        @authors << object
+      when Book
+        @books << object
+      when Reader
+        @readers << object
+      when Order
+        @orders << object
+      end
+    end
+  end
+
+  def generate_library
```

Comment:
This method does not belong to library. Could you please move it to your data generator?